### DISCUSSION TITLE
Testing HTTP get with images/files

### DISCUSSION THREAD

**Discussion by CorvS** [laravel/framework#35918] [Q&A] [answered]
- Laravel Version: 8.22.1
- PHP Version: 7.4.8

`$this->get('https://laravel.com')->assertOk(); // OK`

Runs fine obviously, but if I try the following with an image, that works via browser/curl/postman:

`$this->get('https://laravel.com/img/logotype.min.svg')->assertOk(); // 404`

My test always fails because 404 is returned instead of 200.

Am I doing anything wrong or is this a bug?

**Comment by crynobone** [comment, marked as answer]
Laravel feature tests can only send request to Laravel routes, when you try to access an actual file it would failed. In general if you need to use these type of tests you should be using Laravel Dusk where artisan serve is running under the hood + Chrome.

**Reply by CorvS** [reply]
If the tests can only send to Laravel routes why does it work with "normal" URLs (like laravel.com, google.com) then?

**Reply by crynobone** [reply]
![image](https://user-images.githubusercontent.com/172966/104809407-fb6b1580-5827-11eb-919f-dbf0e04971ca.png)

Is it actually calling the remote URL?

![image](https://user-images.githubusercontent.com/172966/104809433-2b1a1d80-5828-11eb-8fef-a85027800ad4.png)

If you dump the response, it's actually Laravel app own 404.
